os/user/getent_cgo_test.go

```diff
@@ -0,0 +1,95 @@
+// +build aix dragonfly freebsd !android,linux netbsd openbsd solaris !darwin
+// +build cgo,!osusergo
+
+package user
+
+import (
+	"testing"
+)
+
+func TestIterateUser(t *testing.T) {
+	var users = []*User{
+		{Uid: "0", Gid: "0", Username: "root", Name: "System Administrator", HomeDir: "/var/root"},
+		{Uid: "1", Gid: "1", Username: "daemon", Name: "System Services", HomeDir: "/var/root"},
+		{Uid: "4", Gid: "4", Username: "_uucp", Name: "Unix to Unix Copy Protocol", HomeDir: "/var/spool/uucp"},
+		{Uid: "13", Gid: "13", Username: "_taskgated", Name: "Task Gate Daemon", HomeDir: "/var/empty"},
+		{Uid: "24", Gid: "24", Username: "_networkd", Name: "Network Services", HomeDir: "/var/networkd"},
+		{Uid: "25", Gid: "25", Username: "_installassistant", Name: "Install Assistant", HomeDir: "/var/empty"},
+		{Uid: "26", Gid: "26", Username: "_lp", Name: "Printing Services", HomeDir: "/var/spool/cups"},
+		{Uid: "27", Gid: "27", Username: "_postfix", Name: "Postfix Mail Server", HomeDir: "/var/spool/postfix"},
+	}
+
+	err := iuh.usefgetpwent("./testdata/users.txt")
+	if err != nil {
+		t.Errorf("failed to open test file: %v", err)
+	}
+
+	// Test that we retrieve groups in same order as we defined
+	i := 0
+	err = iterateUsers(func(user *User) error {
+		if i < len(users) {
+			if user.Uid != users[i].Uid ||
+				user.Gid != users[i].Gid ||
+				user.Username != users[i].Username ||
+				user.Name != users[i].Name ||
+				user.HomeDir != users[i].HomeDir {
+				print(t)
+				t.Errorf("could not parse users correctly: expected user name: %s got %s \n", user.Name, users[i].Name)
+			}
+		} else {
+			t.Errorf("could not parse users correctly: more user entries in test file than test has defined.")
+		}
+		i++
+		return nil
+	})
+
+	if i != len(users) {
+		t.Errorf("could not parse all users correctly")
+	}
```

Comment:
how about building up a user struct and comparing two slices at the end of the iteration?

Reply:
The effect would still be the same, we would just move the comparison part away from `iterateUsers/iterateGroups`. Also we would need to allocate 2 slices instead of 1. Apart from slightly improved code readability, I don't see any other benefits of this. Am I missing something?